src/main/jbake/content/blog/2017/02/13/monitoring-canary-deployments.adoc

```diff
@@ -0,0 +1,249 @@
+endif::[]
+
+Now, we'll create a new build and new application for our "Blue" deployment. We
+first set the current deployment to have 4 replicas, so that we'll have 80% of
+the traffic going to the old version, and 20% going to the new one.
+
+```bash
+cd account-manager/
+oc scale deploymentconfig account-manager --replicas=4
+oc new-build --binary --name=account-manager-blue -l app=account-manager-blue
+oc start-build account-manager-blue --from-dir=. --follow
+oc new-app \
+  account-manager-blue \
+  -l app=account-manager-blue \
+  HAWKULAR_APM_URI='http://hawkular-apm' \
+  HAWKULAR_APM_USERNAME='admin' \
+  HAWKULAR_APM_PASSWORD='password'
+```
+
+An output similar to this is expected:
+```
+--> Found image 3f61c6a (37 seconds old) in image stream account-manager-blue under tag "latest" for "account-manager-blue"
+
+    * This image will be deployed in deployment config "account-manager-blue"
+    * The image does not expose any ports - if you want to load balance or send traffic to this component
+      you will need to create a service with 'expose dc/account-manager-blue --port=[port]' later
+
+--> Creating resources with label app=account-manager-blue ...
+    deploymentconfig "account-manager-blue" created
+--> Success
+    Run 'oc status' to view your app.
+```
+
+We should also have a new service displayed on OpenShift console. Note the new
+_account-manager-blue_ deployment, with 1 pod.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-02-13-openshift-account-manager-services.png[OpenShift with Blue/Green deployments for Account Manager]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-02-13-openshift-account-manager-services.png[OpenShift with Blue/Green deployments for Account Manager]
+endif::[]
+
+Our Account Manager microservice is a Vert.x Verticle, and reacts on messages sent
+to `AccountManager.getAccount`. As we are using a clustered event bus, our new
+_account-manager-blue_ service will join the cluster and start replying to those
+messages. The Vert.x clustered bus will take care of load balancing the processing
+of the messages for us.
+
+Because of that, we should already have some data being captured and reported to
+our Hawkular APM server. On the _Services_ screen, add the new service _order-manager.account-manager_,
+version _order-manager.account-manager-1_ to the graphic. It should then look similar
+to this:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-02-13-account-manager-blue-added.png[Hawkular APM Services Tab with Blue and Green services]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-02-13-account-manager-blue-added.png[Hawkular APM Services Tab with Blue and Green services]
+endif::[]
+
+We should see the bar for our _blue_ deployment (orange bar on the image) to be
+around 20% of the size of the old version (blue bar on the image). As the number
```

Comment:
Should be 25% - i.e. 20% are new version, 80% are old version, so new version is a quarter of the old version.